securedrop_client/queue.py

```diff
@@ -139,27 +153,38 @@ def __init__(self, api_client: API, session_maker: scoped_session) -> None:
 
         self.main_thread = QThread()
         self.download_file_thread = QThread()
+        self.metadata_thread = QThread()
 
         self.main_queue = RunnableQueue(api_client, session_maker)
         self.download_file_queue = RunnableQueue(api_client, session_maker)
+        self.metadata_queue = RunnableQueue(api_client, session_maker, size=1)
 
         self.main_queue.moveToThread(self.main_thread)
         self.download_file_queue.moveToThread(self.download_file_thread)
+        self.metadata_queue.moveToThread(self.metadata_thread)
 
         self.main_thread.started.connect(self.main_queue.process)
         self.download_file_thread.started.connect(self.download_file_queue.process)
+        self.metadata_thread.started.connect(self.metadata_queue.process)
 
         self.main_queue.paused.connect(self.on_queue_paused)
         self.download_file_queue.paused.connect(self.on_queue_paused)
+        self.metadata_queue.paused.connect(self.on_queue_paused)
+
+        self.main_queue.pinged.connect(self.resume_queues)
+        self.download_file_queue.pinged.connect(self.resume_queues)
+        self.metadata_queue.pinged.connect(self.resume_queues)
```

Comment:
this queue doesn't need to pause or resume so you can remove this as well as the pinged signal. instead you can just call resume_queues from the Controller on_sync_success if the queues are paused.

Reply:
I'm not sure I follow.

The queue decides itself (see the exception handling in the `process` method of the `RunnableQueue` class.) Therefore, the metadata queue could be paused *if* it encounters a problem (which is a good thing). However, the update timer will restart the queue after X period of time, right...?

In which case, I'd argue these signals still need to be there. IYSWIM.

Reply:
So currently, we never stop metadata syncs when they fail because of a request timeout. We want them to run in the background until they succeed, which is why it doesn't make sense to pause the metadata sync queue. It looks like your code will try to enqueue another MetadataSyncJob after 60 seconds but it will be dropped because the queue will be full and paused. The right behavior is to not pause the metadata queue and it makes the code much simpler because you can remove the `pinged` signal